internal-client/src/main/java/org/apache/uniffle/client/factory/ShuffleServerClientFactory.java

```diff
@@ -59,4 +59,9 @@ public synchronized ShuffleServerClient getShuffleServerClient(
     }
     return serverToClients.get(shuffleServerInfo);
   }
+
+  // Only for tests
+  public synchronized void cleanupCache() {
```

Comment:
Is this necessary?  Maybe we could use reflection in our test code. I just have some concern about it.  I don't know the best practice. @LuciferYang Could you give some suggestion?

Reply:
The original intention of this method is to solve the problem of client closed by other test cases. This will make some test cases flaky.

> It just reassigns clients, but doesn't clear the state of the original one, such as close shuffleServerClient

You mean we'd better to close the client when cleaning the cache. Right? @LuciferYang

Reply:
Yes, if we implement such method for `ShuffleServerClientFactory`, it should be more robust. Otherwise, when `cleanupCache` is called,  the resources in previous `clients` may leak?

Reply:
Got your point.

Reply:
Done